src/node_api/core/routes.rs

```diff
@@ -73,6 +88,36 @@ impl Client {
             .await
     }
 
+    /// GET /api/core/v2/info endpoint
+    pub async fn get_node_info(url: &str, auth: Option<NodeAuth>) -> Result<InfoResponse> {
+        let mut url = crate::node_manager::builder::validate_url(Url::parse(url)?)?;
+        if let Some(auth) = &auth {
+            if let Some((name, password)) = &auth.basic_auth_name_pwd {
+                url.set_username(name)
+                    .map_err(|_| crate::Error::UrlAuthError("username".to_string()))?;
```

Comment:
You could change this error to take a static str and not a String, would be easier